src/components/Chart/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useMemo } from 'react';
+import {
+  ResponsiveContainer,
+  BarChart as RechartsBarChart,
+  Bar,
+  Cell,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  LabelList,
+} from 'recharts';
+import max from 'lodash/max';
+import CustomTooltip from './components/CustomTooltip';
+import useTheme from 'hooks/useTheme';
+import CustomLabel from './components/CustomLabel';
+import { getValues, customTickFormatter } from './utils';
+
+const maxYAxisWidth = 220;
+const multiplyFactor = 7.8;
+
+export type HoverInfo = {
+  name: string;
+  value?: string | number;
+  percentage?: string;
+  [prop: string]: number | string | undefined;
+};
+
+type Option = {
+  color?: string;
+  coloringOption?: 'all' | 'bar';
+  hoverInfo?: HoverInfo[];
+};
+
+export type Data = {
+  name: string;
+  subcategory?: string;
+  value?: number;
+  barLabel?: string;
+  options?: Option;
+};
+
+export type Props = {
+  /** This property defines the data to be shown in the Line Chart */
+  data: Data[];
+};
+
+type YAxisProp = {
+  colors: Record<string, string>;
+  y: number;
+  payload: { value: React.ReactNode };
+};
+
+const CustomYAxisTick = ({ colors, y, payload }: YAxisProp) => {
+  const theme = useTheme();
+
+  const fill =
+    typeof payload.value === 'string' && colors[payload.value]
+      ? colors[payload.value]
+      : theme.palette.black;
+
+  return (
+    <text x={0} y={y} textAnchor="start" fill={fill}>
+      <tspan dy="0.335em">{payload.value}</tspan>
+    </text>
+  );
+};
+
+const BarChart: React.FC<Props> = ({ data }) => {
+  const theme = useTheme();
+
+  const barColors = useMemo(() => {
+    return data.map(obj => {
+      if (obj?.options?.color) {
+        return obj?.options?.color;
+      }
+
+      return theme.palette.flat.darkBlue[100];
+    });
+  }, [data, theme.palette.flat.darkBlue]);
+
+  const maxLabelLength = useMemo(() => max(data.map(obj => obj.name.length)), [data]);
```

Comment:
you can refactor and also reuse it to line 86 by using the maxBy lodash method:

example:
```
----  utils/common.js

  const findMaxInData = useCallback(operator => () => maxBy(data, operator), [data]);

  const maxLabelLength = findMaxInData(item => item.name.length);
  const maxValue = findMaxInData(obj => obj.value);

  ```